xapi-client/src/test/java/dev/learning/xapi/client/XapiEnhancedClientTests.java

```diff
@@ -0,0 +1,275 @@
+        """;
+
+    mockWebServer.enqueue(new MockResponse().setStatus("HTTP/1.1 200 OK").setBody(body)
+        .addHeader("Content-Type", "application/json; charset=utf-8"));
+
+    // When Getting StatementIterator
+    final var iterator = client.getStatementIterator().block();
+
+    // Then HasNext Is False
+    assertThat(iterator.hasNext(), is(false));
+
+  }
+
+  @Test
+  void givenEmptyResponseWhenGettingStatementIteratorThenHasNextIsFalse()
+      throws InterruptedException {
+
+    // Given Empty Response
+    // This response is technically invalid by the xApi specification, but we cannot trust the xApi
+    // conformance of the commercial LRSs.
+    final var body = "{}";
+
+    mockWebServer.enqueue(new MockResponse().setStatus("HTTP/1.1 200 OK").setBody(body)
+        .addHeader("Content-Type", "application/json; charset=utf-8"));
+
+    // When Getting StatementIterator
+    final var iterator = client.getStatementIterator().block();
+
+    // Then HasNext Is False
+    assertThat(iterator.hasNext(), is(false));
+
+  }
+
+  @Test
+  void givenEmptyResponseWhenGettingStatementIteratorThenNextThrowsAnException()
+      throws InterruptedException {
+
+    // Given Empty Response
+    final var body = """
+        {
+          "statements" : [
+          ],
+          "more" : ""
+        }
+        """;
+
+    mockWebServer.enqueue(new MockResponse().setStatus("HTTP/1.1 200 OK").setBody(body)
+        .addHeader("Content-Type", "application/json; charset=utf-8"));
+
+    // When Getting StatementIterator
+    final var iterator = client.getStatementIterator().block();
+
+    // Then Next Throws An Exception
+    assertThrows(NoSuchElementException.class, () -> iterator.next());
+
+  }
+
+  @Test
+  void whenVoidingStatementThenBodyIsExpected() throws InterruptedException {
+
+    mockWebServer.enqueue(new MockResponse().setStatus("HTTP/1.1 200 OK")
+        .setBody("[\"2eb84e56-441a-492c-9d7b-f8e9ddd3e15d\"]")
+        .addHeader("Content-Type", "application/json"));
+
+    final var attemptedStatement = Statement.builder()
+
+        .id(UUID.fromString("175c9264-692f-4108-9b7d-0ba64bd59ac3"))
+
+        .agentActor(a -> a.name("A N Other").mbox("mailto:another@example.com"))
+
+        .verb(Verb.ATTEMPTED)
+
+        .activityObject(o -> o.id("https://example.com/activity/simplestatement")
+            .definition(d -> d.addName(Locale.ENGLISH, "Simple Statement")))
+
+        .build();
+
+    // When Voiding Statement
+    client.voidStatement(attemptedStatement).block();
+
+    final var recordedRequest = mockWebServer.takeRequest();
+
+    // Then Body Is Expected
+    assertThat(recordedRequest.getBody().readUtf8(), is(
+        "{\"actor\":{\"objectType\":\"Agent\",\"name\":\"A N Other\",\"mbox\":\"mailto:another@example.com\"},\"verb\":{\"id\":\"http://adlnet.gov/expapi/verbs/voided\",\"display\":{\"und\":\"voided\"}},\"object\":{\"objectType\":\"StatementRef\",\"id\":\"175c9264-692f-4108-9b7d-0ba64bd59ac3\"}}"));
```

Comment:
Add a test for an iterated transfer into a stream and limited. First response should have 2 results seciond response should have 1 result. there should only be one call.